llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -2776,6 +2776,12 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
       }
       return Cost;
     }
+    case Intrinsic::maximumnum:
+    case Intrinsic::minimumnum: {
+      if (TLI->isOperationLegalOrPromote(llvm::ISD::FMAXNUM_IEEE, LT.second))
+        return LT.first * 3;
```

Comment:
Why `* 3`?

Would be good to have dedicated cost model tests as well

Reply:
For Architectures that support `FMAXNUM_IEEE`, the `maximumnum` is implemented by:

```
ARG0 = FMAXNUM_IEEE  ARG0, ARG0  // to queit sNaN
ARG1 = FMAXNUM_IEEE  ARG1, ARG1  // to queit sNaN
RESULT= FMAXNUM_IEEE  ARG0, ARG1  // do real max
```

Reply:
Can you comment on the expansion here? 

We can't have a dedicated test for FMAXNUM_IEEE. As implemented, it's a backend only implementation detail. However the langref now states it's the same as llvm.maxnum, so eventually this will be equivalent